gourmet/tests/test_plugin_loader.py

```diff
@@ -1,37 +1,45 @@
 import unittest
 import tempfile
 import os
+import shutil
 
-import gourmet.gglobals
+from gourmet import gglobals
 
-# clear out Gourmet's DB
-tmpdir = tempfile.mktemp()
-os.makedirs(tmpdir)
-gourmet.gglobals.gourmetdir = tmpdir
 
-import gourmet.GourmetRecipeManager
-import gourmet.backends.db
+class Test (unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # create a temporary test directory
+        cls.tmp_dir = tempfile.mkdtemp(".pl")
+        gglobals.gourmetdir = cls.tmp_dir
 
-gourmet.backends.db.RecData.__single = None
-gourmet.GourmetRecipeManager.GourmetApplication.__single = None
-# end clearing out code
+        # continue to import with 'gourmetdir' set to 'tmp_dir', otherwise 'gourmetdir' is set to '~/.gourmet'
+        from gourmet.plugin_loader import get_master_loader  # noqa: E402 import not at top of file
 
-from gourmet.plugin_loader import get_master_loader
+        cls.ml = get_master_loader()
 
-class Test (unittest.TestCase):
+    @classmethod
+    def tearDownClass(cls):
+        # delete temporary test directory
+        try:
+            shutil.rmtree(cls.tmp_dir)
+        except OSError as e:
+            print("Error: {} : {}").format(cls.tmp_dir, e.strerror)
 
     def testDefaultPlugins (self):
-        ml = get_master_loader()
-        ml.load_active_plugins()
-        print('active:',ml.active_plugins)
-        print('instantiated:',ml.instantiated_plugins)
-        assert(not ml.errors)
+        self.ml.load_active_plugins()
+        print('active:',self.ml.active_plugins)
+        print('instantiated:',self.ml.instantiated_plugins)
+        self.assertEqual(len(self.ml.errors), 0)  # there should be 0 plugin errors
 
     def testAvailablePlugins (self):
-        ml = get_master_loader()
-        for st in ml.available_plugin_sets:
-            if st not in ml.active_plugins:
-                ml.activate_plugin_set(st)
+        # search module directories for available plugins
+        for module_name, plugin_set in self.ml.available_plugin_sets.items():
+            if module_name not in self.ml.active_plugin_sets:
+                self.ml.activate_plugin_set(plugin_set)
+        self.ml.save_active_plugins()
+        self.assertTrue(os.path.exists(self.ml.active_plugin_filename))
```

Comment:
```suggestion
        assert os.path.exists(self.ml.active_plugin_filename)
```

Feel free to ignore this. But I've never seen the point of the `.assertTrue()` method. Things like `.assertEqual()` give more informative error messages by knowing about what they're checking, but `.assertTrue()` can't say anything interesting.

(I also made a library [testpath](https://pypi.org/project/testpath/) which includes things like `assert_isfile()`, but it's probably not worth adding just for one assertion.)

Reply:
I've removed the `assertTrue`. Your `testpath` library look thorough, as you mentioned above '_probably not worth adding just for one assertion_'.